gdc/dependencies.go

```diff
@@ -0,0 +1,153 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2017 RightScale, Inc, All Rights Reserved Worldwide.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package main
+
+import (
+	"fmt"
+	"go/parser"
+	"go/token"
+	"log"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+)
+
+func printImports(path string) {
+	// Print the imports from the file's AST.
+	for _, s := range getFileImports(path) {
+		fmt.Println(s)
+	}
+}
+
+func getFileImports(path string) []string {
+	fset := token.NewFileSet() // positions are relative to fset
+	f, err := parser.ParseFile(fset, path, nil, parser.ImportsOnly)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	imports := make([]string, len(f.Imports))
+	for i, s := range f.Imports {
+		imports[i] = s.Path.Value
+	}
+	return imports
+}
+
+func getGoFiles(path string) []string {
+	fileList := []string{}
+	err := filepath.Walk(path, func(path string, f os.FileInfo, err error) error {
+		if strings.HasSuffix(path, ".go") {
+			fileList = append(fileList, path)
+		}
+		return nil
+	})
+
+	if err != nil {
+		log.Fatal(err)
+	}
+	return fileList
+}
+
+func getAllFiles(path string) []string {
+	fileList := []string{}
+	err := filepath.Walk(path, func(path string, f os.FileInfo, err error) error {
+		fileList = append(fileList, path)
+		return nil
+	})
+
+	if err != nil {
+		log.Fatal(err)
+	}
+	return fileList
+}
+
+func getExecutionDir() string {
+	// currentDir, err := filepath.Abs(filepath.Dir(os.Args[0]))
+	currentDir, err := os.Getwd()
+	if err != nil {
+		log.Fatal(err)
+	}
+	return currentDir
+}
+
+func getSortedKeys(aMap map[string]struct{}) []string {
+	keys := make([]string, len(aMap))
+
+	i := 0
+	for k := range aMap {
+		keys[i] = k
+		i++
+	}
+	sort.Strings(keys)
+	return keys
+}
+
+func getImports(directory string) (files []string) {
```

Comment:
BTW, I'm not sure if this is perfectly equivalent to your code, but here's how I've gotten a list of every dependent package, from bash:
```
go list -f '{{join .Deps "\n"}}'
```
I'm curious if that yields an equivalent list..